tests/test_api_v1.py

```diff
@@ -1,7 +1,80 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 import json
 
+import pytest
+
 
 def test_ping(client):
     rv = client.get('/api/v1/ping')
     assert json.loads(rv.data.decode('utf-8')) is True
+
+
+def test_create_and_fetch_request(client, db):
+    data = {
+        'git_repo': 'https://github.com/release-engineering/retrodep.git',
+        'git_ref': 'c50b93a32df1c9d700e3e80996845bc2e13be848',
+        'pkg_managers': ['gomod']
+    }
+
+    rv = client.post('/api/v1/requests', json=data)
+    assert rv.status_code == 201
+    created_request = json.loads(rv.data.decode('utf-8'))
+    for key, expected_value in data.items():
+        assert expected_value == created_request[key]
+
+    request_id = created_request['id']
+    rv = client.get('/api/v1/requests/{}'.format(request_id))
+    assert rv.status_code == 200
+    fetched_request = json.loads(rv.data.decode('utf-8'))
+
+    assert created_request == fetched_request
+
+
+def test_missing_request(client, db):
+    rv = client.get('/api/v1/requests/1')
+    assert rv.status_code == 404
+
+
+def test_malformed_request_id(client, db):
+    rv = client.get('/api/v1/requests/spam')
+    assert rv.status_code == 400
+    error_msg = json.loads(rv.data.decode('utf-8'))['error']
+    assert 'invalid literal' in error_msg
```

Comment:
This seems like a scary message that should be improved

Reply:
Changed it to "X is not a valid request ID"